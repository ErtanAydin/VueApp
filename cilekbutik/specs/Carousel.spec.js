import { mount, RouterLinkStub } from "@vue/test-utils";
import Carousel from "@/components/Carousel";

describe("Carousel", () => {
  test("mounts properly", () => {
    const wrapper = Carousel;
    expect(wrapper).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = mount(Carousel, { 
      stubs: {
        NuxtLink: RouterLinkStub
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe('next', () => {
  test('next emit test', () => {
    const wrapper = mount(Carousel, { 
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    wrapper.vm.next()
    expect(wrapper.emitted().next[0]).toEqual([])
  })
})

describe('prev', () => {
  test('prev emit test', () => {
    const wrapper = mount(Carousel, { 
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    wrapper.vm.prev()
    expect(wrapper.emitted().prev[0]).toEqual([])
  })
})