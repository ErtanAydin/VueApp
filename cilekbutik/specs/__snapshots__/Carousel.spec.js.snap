// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders properly 1`] = `<div class="carousel"> <button class="next carousel-button" style="font-size: x-large;"><i id="normal-text" class="fas fa-chevron-right"></i></button> <button class="prev carousel-button" style="font-size: x-large;"><i id="normal-text" class="fas fa-chevron-left"></i></button></div>`;
